docs/sources/developers/plugins/add-support-for-variables.md

```diff
@@ -79,6 +79,28 @@ SELECT * FROM services WHERE id IN (admin,auth,billing)
 
 For more information on the available variable formats, refer to [Advanced variable format options](https://grafana.com/docs/grafana/latest/variables/advanced-variable-format-options/).
 
+## Set a variable from your plugin
+
+Not only can you read the value of a variable, you can also update the variable from your plugin. Use [LocationSrv.update()]({{< relref "../../packages_api/runtime/locationsrv.md/#update-method" >}}) to update a variable using query parameters.
+
+The following example updates the current value of a variable called `service` to `billing`. By setting `partial` to `true`, you leave the other query parameters unchanged. `replace` determines whether to update the current URL state, or to create a new history entry. We recommend setting both to `true` when updating variables.
```

Comment:
This is a little confusing. So, the variable name is `service`, and its value is being updated to `billing`?

All the code formatting in this paragraph and indirect language make it a bit hard to follow. Consider rewriting everything after the first sentence as numbered steps.

Reply:
Yes. Not really sure how I can make it clearer. Maybe leave out the value part?

> The following example shows how to update a variable called `service`.

The only instruction is more or less to copy the code snippet. The rest are not really instructions, but explains what the properties do. I would've liked to put the "By setting ..." after the code block, but I seem to remember from earlier feedback that details of an example in the paragraph should go before the example rather than after.

Reply:
Yes, that would help